src/elements/MetricsSelector/MetricsSelector.js

```diff
@@ -119,25 +134,31 @@ const MetricsSelector = ({ maxSelectionNumber, metrics, name, onSelect, preselec
   }, [windowClickHandler, windowScrollHandler, isOpen])
 
   const handleOnChange = selectedMetrics => {
-    onSelect(
-      selectedMetrics.map(metricFullName => {
-        return metrics.find(metric => metric.full_name === metricFullName)
-      })
-    )
+    setPreAppliedMetrics(selectedMetrics)
+  }
+
+  const handleApply = () => {
+    const matchedMetrics = preAppliedMetrics.map(metricFullName => {
+      return metrics.find(metric => metric.full_name === metricFullName)
+    })
+    onSelect(matchedMetrics)
+    setAppliedMetrics(matchedMetrics)
+    formRef.current.change('metrics', preAppliedMetrics)
```

Comment:
delete this usage of the `change` method. It's redundant

Reply:
@mavdryk done.